### PR TITLE
Ignore unknown members when one of the inferred owners is ignored for typechecks

### LINKED ISSUE #1276: False positive on no-member for inferred classes that are ignored
### Steps to reproduce
1. get some code that end up having an instance that can be one of multiple classes, at least one of them being ignored for typecheck (using `ignored-modules` or `ignored-classes`), then use a non existent member on this instance. E.g.[`schroedinger.py` example file](https://gist.github.com/pcorpet/a0eaa282754a95273bccacd07419d1f7)
2. run the commandline with pylint and make sure you ignore one of the classes (here I've added extra commandline flags to make sure nothing interferes with the result) 
```sh
pylint --rcfile=/dev/null --ignored-classes=MyVeryOwnError --reports=no --disable=R schroedinger.py
```

### Current behavior
A `no-member` error is shown for the culprit line and mentions only the inferred classes that are not ignored.
```
E: 26,14: Instance of 'ValueError' has no 'node' member (no-member)
```

### Expected behavior
No error at all.

### pylint --version output
```
pylint 1.6.4, 
astroid 1.4.8
Python 3.5.2 (default, Nov 17 2016, 22:42:56) 
[GCC 4.9.2]
```

### My Guess

I believe that what happens is that the [continue](https://github.com/PyCQA/pylint/blob/69f480dfe698b17756f97ca61cd7c6390e3e7f66/pylint/checkers/typecheck.py#L627) statement which covers for ignored owners should validate the member instead of trusting other inferred owers to know better. So I would replace it by a `break`. Any thoughts?

### PR DESCRIPTION
**Pull request by pcorpet**
### Fixes / new features
Fixes #1276.

### CODE REVIEW

**Comment by pcorpet** [comment]
@rogalski done, checks in progress.

File: ChangeLog [hunk 1]
```diff
@@ -343,6 +343,10 @@ Release date: 2017-04-13
 
       Fixes #59
 
+    * Ignore unknown members when one of the inferred owners is ignored for
```
Comment (pcorpet) [inline ChangeLog]: I will move this to the latest part of the Changelog once this PR gets some attention: no need to chase after the versions before that.

**Comment by PCManticore** [comment]
In general, I'm happy with the idea, but what I would like to see is a flag through which this can be controlled, defaulting to the behavior that you are introducing in this PR. Say something along the lines of ``skip-on-ignored-owner``, with the default value set to ``True``. This way, folks could still detect other problems for other branches of inference by deactivating the flag.

**Comment by pcorpet** [comment]
Sorry for the long delay. @PCManticore I'm not sure who would want to deactivate that flag but if you think that's the right move, I'll just do that.

**Comment by PCManticore** [comment]
@pcorpet I'm closing this PR for now in an effort to clean our PR tracker from old pull requests. As I mentioned earlier, I'm happy to have this in pylint, albeit as long as it comes with that flag. Feel free to resubmit a new PR when you get a chance.

**Comment by pcorpet** [comment]
Thanks for the offer. I do think that my fix is/was the right one and that no-one would actually want the no-member error to trigger. As I'm almost sure very few people would notice nor use that flag I'm reluctant to add it, so I'll leave this PR closed. However the bug #1276 is still open and if someone wants to do what's suggested above, I still believe it's the right way to go.

**Pull request closed without merging**